ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -133,10 +141,13 @@ bracketChainSyncClient tracer ChainDbView { getIsInvalidBlock } varCandidates
       withWatcher
         "ChainSync.Client.rejectInvalidBlocks"
         (invalidBlockWatcher varCandidate)
-        $ body varCandidate
+        $ body (atomically . writeTVar varCandidate)
   where
     newCandidateVar = do
-      varCandidate <- newTVarIO $ AF.Empty AF.AnchorGenesis
+      varCandidate <- newTVarIO $ CandidateFragment
+        { candidateChain        = AF.Empty AF.AnchorGenesis
+        , candidateIsLowDensity = False
```

Comment:
Better name for `candidateIsLowDensity`. eg `candidateInForecastRange` (ie negated)

Reply:
Actually, we can be more specific even: `candidateDiscardIfNotPlausible`.